codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/ServiceClientGenerator.kt

```diff
@@ -67,11 +67,12 @@ class ServiceClientGenerator(private val ctx: RenderingContext<ServiceShape>) {
         requireNotNull(ctx.shape) { "ServiceShape is required to render a service client" }
     private val serviceSymbol = ctx.symbolProvider.toSymbol(service)
     private val writer = ctx.writer
+    private val visibility = ctx.settings.build.visibility.serviceClient
 
     fun render() {
         writer.write("\n\n")
-        writer.write("public const val ServiceId: String = #S", ctx.settings.sdkId)
-        writer.write("public const val SdkVersion: String = #S", ctx.settings.pkg.version)
+        writer.write("$visibility const val ServiceId: String = #S", ctx.settings.sdkId)
+        writer.write("$visibility const val SdkVersion: String = #S", ctx.settings.pkg.version)
```

Comment:
**Nit**: Mix templating styles makes codegen harder to read. I recommend using `#L` instead of `$visibility`. Applies to other changes in this PR.